### PR TITLE
Inject reserved role name checker into role mgmt actions

### PR DESCRIPTION
**Pull request by n1v0lg**
This PR makes reserved role name validation injectable for the create and delete roles actions.

### CODE REVIEW

**Comment by n1v0lg** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Comment by n1v0lg** [comment]
Flipping this back to draft since I'll use today to also fold in the Put Roles changes.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/DeleteRoleRequest.java [hunk 1]
```diff
@@ -19,27 +20,28 @@
...
-        name = in.readString();
-        refreshPolicy = RefreshPolicy.readFrom(in);
-    }
+    private boolean restrictRequest;
 
     public DeleteRoleRequest() {}
 
-    @Override
-    public DeleteRoleRequest setRefreshPolicy(RefreshPolicy refreshPolicy) {
+    public DeleteRoleRequest setRefreshPolicy(@Nullable String refreshPolicy) {
```
Comment (n1v0lg) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/role/DeleteRoleRequest.java:32]: This is part of making the action local part of which is removing `implements WriteRequest`

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Review by slobodanadamovic — APPROVED** [review]
LGTM

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Pull request merged**